sdk/core/Azure.Core.Experimental/src/Auth/Interfaces/IScopedContext.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel;
+
+/// <summary>
+/// Represents a provider that can provide a token.
+/// </summary>
+public interface IScopedFlowContext : ITokenContext
+{
+    /// <summary>
+    /// Gets the scopes required to authenticate.
+    /// </summary>
+    string[] Scopes { get; }
+
+    /// <summary>
+    /// Clones the current context with additional scopes.
+    /// </summary>
+    /// <param name="additionalScopes"></param>
+    /// <returns></returns>
+    object CloneWithAdditionalScopes(string[] additionalScopes);
```

Comment:
Should this just be `.WithAdditionalScopes`?  It looks like `.With` is fairly popular.  https://learn.microsoft.com/en-us/dotnet/api/?term=With&view=net-8.0

Reply:
And is it worth a `params` here?

Reply:
I like it.